mParticle_AppboyTests/mParticle_AppboyTests.m

```diff
@@ -353,4 +354,55 @@ - (void)testlogCommerceEvent {
     [mockClient stopMocking];
 }
 
+- (void)testTypeDetection {
+    MPKitAppboy *kit = [[MPKitAppboy alloc] init];
+    
+    Appboy *testClient = [[Appboy alloc] init];
+    id mockClient = OCMPartialMock(testClient);
+    [kit setAppboyInstance:mockClient];
+        
+    XCTAssertEqualObjects(mockClient, [kit appboyInstance]);
+    
+    
+    MPEvent *event = [[MPEvent alloc] initWithName:@"test event" type:MPEventTypeNavigation];
```

Comment:
You might want to add a date and negative integer value here to cover all your bases.